cmd/kubeadm/app/util/version.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+const (
+	kubeReleaseBucketURL    = "https://storage.googleapis.com/kubernetes-release"
+	kubeDevReleaseBucketURL = "https://storage.googleapis.com/kubernetes-release-dev"
+)
+
+var (
+	kubeReleaseRegex        = regexp.MustCompile(`^v(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)([-0-9a-zA-Z_\.+]*)?$`)
+	kubeReleaseLabelRegex   = regexp.MustCompile(`^release/[-\w_\.]+`)
```

Comment:
Could we just allow `release/(stable|latest)-[\.\d]+` or something here?
Right now I feel that for example `release/foobar-1.7` could be passed to it just fine and the validation will pass